bin/test/main.ml

```diff
@@ -294,62 +294,63 @@ let run_exn (`Setup ()) (`Non_deterministic non_deterministic)
     let active =
       active t && Block.version_enabled t && (not (Block.skip t))
     in
-    match active, non_deterministic, Block.mode t, Block.value t with
+    let print_block () = Block.pp ?syntax ppf t in
+    match active, 
+          non_deterministic, 
+          Block.file t, 
+          Block.mode t, 
+          Block.value t 
+    with
```

Comment:
This match tries to do too much things.

What do you think of:
- Moving the last 2 cases to functions
  The patterns on `Block.file t` can be moved to a nested match
- Do the `active` check with an `if`
  The first `Error _` case is not necessary, it is handled again in the 2 last cases

So this monster-match now only do the non-det checks.

Reply:
We've discussed this offline with @voodoos and decided to focus on the feature and keep the deeper refactoring for later, there's so much that's wrong about this function I'd prefer to keep a clear separation between refactoring and feature addition here.